Essentials/src/com/earth2me/essentials/Trade.java

```diff
@@ -289,100 +302,78 @@ public BigDecimal getCommandCost(final IUser user) {
-            }
-        }
-        sb.append(",\"");
-        if (receiver != null) {
-            sb.append(receiver);
-        }
-        sb.append("\",");
-        if (pay == null) {
-            sb.append("\"\",\"\",\"\"");
-        } else {
-            if (pay.getItemStack() != null) {
-                sb.append(pay.getItemStack().getAmount()).append(",");
-                sb.append(pay.getItemStack().getType().toString()).append(",");
-                sb.append(pay.getItemStack().getDurability());
-            }
-            if (pay.getMoney() != null) {
-                sb.append(pay.getMoney()).append(",");
-                sb.append("money").append(",");
-                sb.append(ess.getSettings().getCurrencySymbol());
-            }
-            if (pay.getExperience() != null) {
-                sb.append(pay.getExperience()).append(",");
-                sb.append("exp").append(",");
-                sb.append("\"\"");
-            }
-        }
-        if (loc == null) {
-            sb.append(",\"\",\"\",\"\",\"\"");
-        } else {
-            sb.append(",\"");
-            sb.append(loc.getWorld().getName()).append("\",");
-            sb.append(loc.getBlockX()).append(",");
-            sb.append(loc.getBlockY()).append(",");
-            sb.append(loc.getBlockZ()).append(",");
-        }
-        sb.append("\n");
+
+        // [Time] [Type:Subtype:Event]
+        String logFormat = "[%s] %s:%s:%s send:%s/rec:%s/pay:%s/charge:%s/loc:%s \n";
+
+        String time = DateFormat.getDateTimeInstance(DateFormat.FULL, DateFormat.FULL).format(new Date());
+
+        String safeSender = sender != null ? sender : "??";
+        String safeReceiver = receiver != null ? receiver : "??";
+
+        String locationString = loc != null
+                ? String.format("%s:%d,%d,%d", loc.getWorld().getName(), loc.getBlockX(), loc.getBlockY(),
+                        loc.getBlockZ())
+                : "??";
+
+        String log = String.format(logFormat, time, type, subtype, event, safeSender, safeReceiver,
+                formatTrade(pay, ess), formatTrade(charge, ess), locationString);
+
         try {
-            fw.write(sb.toString());
+            fw.write(log);
             fw.flush();
         } catch (IOException ex) {
             Logger.getLogger("Essentials").log(Level.SEVERE, null, ex);
         }
     }
 
+    private static String formatTrade(Trade trade, IEssentials ess) {
+        if (trade == null)
+            return "??";
+        else if (trade.getItemStack() != null)
+            return String.format("(amt:%d,type:%s)", trade.getItemStack().getAmount(),
+                    trade.getItemStack().getType().name());
+        else if (trade.getMoney() != null)
+            return String.format("(amt:%f,money,cur:%s)", trade.getMoney(), ess.getSettings().getCurrencySymbol());
+        else if (trade.getExperience() != null)
+            return String.format("(amt:%d, exp)", trade.getExperience());
```

Comment:
```suggestion
            return String.format("(amt:%d,exp)", trade.getExperience());
```
I think the space here might be a typo (?)

Reply:
Yes, I overlooked that